src/cli/utils.ts

```diff
@@ -0,0 +1,186 @@
+  if (num === 0) {
+    return []
+  }
+
+  const optionsWithRandomness = options.map((o) => {
+    return { option: o, sortkey: Math.random() } as objectWithSortkey<T>
+  })
+  optionsWithRandomness.sort((first, second) =>
+    first.sortkey < second.sortkey ? -1 : 1
+  )
+  return optionsWithRandomness.map((or) => or.option).slice(0, num)
+}
+
+interface GenerateCVRParams {
+  election: Election
+  precinctId: string
+  ballotStyleId: string
+  hasOvervote?: boolean
+  hasUndervote?: boolean
+  hasWriteins?: boolean
+  fromMultipage?: boolean
+  isTestBallot?: boolean
+  ballotType?: BallotType
+}
+
+export const generateCVR = ({
+  election,
+  precinctId,
+  ballotStyleId,
+  isTestBallot = true,
+  hasOvervote = false,
+  hasUndervote = false,
+  hasWriteins = false,
+  fromMultipage = false,
+}: // TODO: include ballot type?
+GenerateCVRParams): CastVoteRecord | undefined => {
+  const ballotStyle = getBallotStyle({ election, ballotStyleId })
+  const precinct = getPrecinctById({ election, precinctId })
+
+  if (!ballotStyle || !precinct) {
+    return
+  }
+
+  const votes: Dictionary<string[]> = {}
+
+  // get all the contests
+  const contests = getContests({ ballotStyle, election })
+
+  if (contests.length === 0) return
+
+  const overvoteContestId = hasOvervote
+    ? pickAtRandom(contests, 1)[0].id
+    : undefined
+
+  // let's not do too many undervotes, 1/3 at most
+  const numUndervotes = hasUndervote
+    ? randomIntegerLessThan(contests.length / 3)
+    : 0
+  const undervoteContestIds = hasUndervote
+    ? pickAtRandom(contests, numUndervotes).map((c) => c.id)
+    : []
+
+  // let's do even fewer write-ins, 1/5 at most
+  const contestsWithWriteins = contests.filter(
+    (c) => c.type === 'candidate' && (c as CandidateContest).allowWriteIns
+  )
+  const numWriteins = hasWriteins
+    ? randomIntegerLessThan(contestsWithWriteins.length / 5)
+    : 0
+  const writeinContestIds = hasWriteins
+    ? pickAtRandom(contestsWithWriteins, numWriteins).map((c) => c.id)
+    : []
+
+  // if multipage, pick a page
+  const contestsPerPage = 10
+  let pageNumber
+  if (fromMultipage) {
+    const numPages = Math.ceil(contests.length / contestsPerPage)
+    pageNumber = randomIntegerLessThan(numPages) + 1
+  }
+
+  const filteredContests = pageNumber
+    ? contests.slice(
+        contestsPerPage * (pageNumber - 1),
+        contestsPerPage * pageNumber
+      )
+    : contests
+
+  filteredContests.forEach((contest) => {
+    // overvote takes precedence on undervote
+    const contestIsOvervote = overvoteContestId === contest.id
+    const contestIsUndervote =
+      !contestIsOvervote && undervoteContestIds.includes(contest.id)
+
+    const contestHasWritein = writeinContestIds.includes(contest.id)
+
+    let options: string[]
+    let seats: number
+    switch (contest.type) {
+      case 'candidate':
+        options = (contest as CandidateContest).candidates.map((c) => c.id)
+        seats = (contest as CandidateContest).seats
```

Comment:
```suggestion
        options = contest.candidates.map((c) => c.id)
        seats = contest.seats
```

Cast isn't necessary.